File: Python/Projects/SecondLargestNum.py
class secondLargestNum:
    def largestNum(self, a):
        max1 = -float('inf')
        max2 = -float('inf')
        for num in a:
            if num > max1:
                max2 = max1
                max1 = num
            elif num > max2 and num != max1:
                max2 = num

        if max2 == -float('inf'):
            print("No second largest number found")
        else:
            print(f"Second largest number is: {max2}")

def main():
    numbers = []
    print("Enter numbers or 'e' to exit:")

    while True:
        userInputNum = input(f"# {len(numbers)}: ").strip()

        if userInputNum == "e":
            break

        try:
            num = int(userInputNum)
            numbers.append(num)
        except ValueError:
            print("Invalid value, Enter 'e' to exit")

    fsln = secondLargestNum()
    fsln.largestNum(numbers)

if __name__  == "__main__":
    main()

File: Java/Practice/SecondLargestNum.java
import java.util.ArrayList;
import java.util.Scanner;

class FindSecondLargestNum{

    void nums(ArrayList<Integer> a){
        int max1 = Integer.MIN_VALUE;
        int max2 = Integer.MIN_VALUE;

        for(int num : a){
            if(num > max1){
                max2 = max1;
                max1 = num;
            } else if(num > max2 && num != max1){
                max2 = num;
            }
        }
        if (max2 == Integer.MIN_VALUE) {
            System.out.println("No second largest number found");            
        } else {
            System.out.println("Second largest number is: " + max2);
        }
    }

}

public class SecondLargestNum {
    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        ArrayList<Integer> numbers = new ArrayList<>();
        int counter = 1;

        while (true) {
            System.out.println("Enter the number or 'e' to exit: " + counter);
            if (scn.hasNextInt()) {
                numbers.add(scn.nextInt());
                counter++;
            } else {
                String input = scn.next();
                if (input.equalsIgnoreCase("e")) {
                    break;
                } else {
                    System.out.println("Invalid, Enter 'e' to exit");
                }
            }
        }
        FindSecondLargestNum fsln = new FindSecondLargestNum();
        fsln.nums(numbers);     

        scn.close();
    }
}
